File: scripts/test_solutions/684_redundant_connection.py
def findRedundantConnection(edges):
    # Use max node value to size parent array (1-indexed nodes)
    max_node = max(max(a, b) for a, b in edges)
    parent = list(range(max_node + 1))

    def find(x):
        if parent[x] != x:
            parent[x] = find(parent[x])
        return parent[x]

    for a, b in edges:
        pa, pb = find(a), find(b)
        if pa == pb:
            return [a, b]
        parent[pa] = pb
    return []

File: scripts/test_solutions/684_redundant_connection.js
var findRedundantConnection = function(edges) {
    const parent = Array(edges.length + 1).fill(null).map((_, i) => i);
    function find(x) { if (parent[x] !== x) parent[x] = find(parent[x]); return parent[x]; }
    for (const [a, b] of edges) {
        const pa = find(a), pb = find(b);
        if (pa === pb) return [a, b];
        parent[pa] = pb;
    }
};
